tests/e2e/__testers__.ts

```diff
@@ -4,29 +4,47 @@ import { values } from 'lodash'
 import stripAnsi from 'strip-ansi'
 import { ClientBase } from '@prisma-spectrum/reflector/dist-cjs/Client'
 import { Reflector } from '@prisma-spectrum/reflector'
+import { getPrismaClient } from '~/tests/e2e/helpers/getPostgresAdminPrismaClient'
+import { PrismaClient } from '@prisma/client'
 
-export const testTemplate = (params: {
+export interface DBTestParams {
   templateName: PrismaTemplates.$Types.Template['_tag']
-  /** A string or regex pattern of the expected output (stdout) when the project dev mode is run.*/
   expectedDevOutput: RegExp | string
-}) => {
-  jest.setTimeout(50_000)
+  datasourceProvider: Reflector.Schema.DatasourceProviderNormalized
+  connectionStringBase: string
+  getPrismaAdmin: getPrismaClient
+  databaseActions: {
+    resetDatabase: (prismaClient: PrismaClient, databaseName: string) => Promise<void>
+    initDatabase: (prismaClient: PrismaClient, databaseName: string) => Promise<void>
+  }
+  prismaConfig?: {
+    referentialIntegrity?: 'prisma' | 'foreignKeys'
+  }
+}
```

Comment:
Exposing configuration for `datasourceProvider` and the `prismaConfig` should be all that's needed here.

Also, both of those configurations fall within the template confirmation. 

Test template is about testing templates. It should be exposing the template config, or subset of it, for the testing, plus whatever other meta aspects of test validation are needed (e.g. `expectations` below).

So the parameters I see us needing are something like this:

```ts
export const testTemplate = (params: {
  /** Test template configuration */
  input: {
    /** Test template name */
    name: PrismaTemplates.$Types.Template['_tag']
    datasourceProvider: Reflector.Schema.DatasourceProviderNormalized
    referentialIntegrity?: ...
  }
  expectations: {
    /** A string or regex pattern of the expected output (stdout) when the project dev mode is run.*/
    devOutput: RegExp | string
  }
}

```

No other input should  be needed from the test case side.

Reply:
Correction on the above, the testTemplate does need to know the connection string _as well_ since that is not part of the template interface.

Reply:
I assume that will be inside a switch case. I'll refactor that as well. But still worried about the devs coming to this project in the future and adding new DB tests.